x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorSpecTests.java

```diff
@@ -0,0 +1,103 @@
+package org.elasticsearch.xpack.sql.planner;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.ql.TestUtils;
+import org.elasticsearch.xpack.ql.index.EsIndex;
+import org.elasticsearch.xpack.ql.index.IndexResolution;
+import org.elasticsearch.xpack.ql.plan.logical.LogicalPlan;
+import org.elasticsearch.xpack.ql.type.EsField;
+import org.elasticsearch.xpack.sql.SqlTestUtils;
+import org.elasticsearch.xpack.sql.analysis.analyzer.Analyzer;
+import org.elasticsearch.xpack.sql.analysis.analyzer.Verifier;
+import org.elasticsearch.xpack.sql.expression.function.SqlFunctionRegistry;
+import org.elasticsearch.xpack.sql.optimizer.Optimizer;
+import org.elasticsearch.xpack.sql.parser.SqlParser;
+import org.elasticsearch.xpack.sql.plan.physical.EsQueryExec;
+import org.elasticsearch.xpack.sql.plan.physical.PhysicalPlan;
+import org.elasticsearch.xpack.sql.stats.Metrics;
+import org.elasticsearch.xpack.sql.types.SqlTypesTests;
+import org.elasticsearch.xpack.sql.util.DateUtils;
+import org.hamcrest.Matcher;
+import org.junit.BeforeClass;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Map;
+
+public class QueryTranslatorSpecTests extends ESTestCase {
+
+    private static final String TEST_FILENAME = "querytranslator_tests.txt";
+
+    private static class TestContext {
+        private final SqlParser parser;
+        private final Analyzer analyzer;
+        private final Optimizer optimizer;
+        private final Planner planner;
+
+        TestContext(String mappingFile) {
+            parser = new SqlParser();
+            Map<String, EsField> mapping = SqlTypesTests.loadMapping(mappingFile);
+            EsIndex test = new EsIndex("test", mapping);
+            IndexResolution getIndexResult = IndexResolution.valid(test);
+            analyzer = new Analyzer(SqlTestUtils.TEST_CFG, new SqlFunctionRegistry(), getIndexResult, new Verifier(new Metrics()));
+            optimizer = new Optimizer();
+            planner = new Planner();
+        }
+
+        LogicalPlan plan(String sql, ZoneId zoneId) {
+            return analyzer.analyze(parser.createStatement(sql, zoneId), true);
+        }
+
+        LogicalPlan plan(String sql) {
+            return plan(sql, DateUtils.UTC);
+        }
+
+        PhysicalPlan optimizeAndPlan(String sql) {
+            return optimizeAndPlan(plan(sql));
+        }
+
+        PhysicalPlan optimizeAndPlan(LogicalPlan plan) {
+            return planner.plan(optimizer.optimize(plan),true);
+        }
+    }
+
+    private static TestContext testContext;
+
+    @BeforeClass
+    public static void init() {
+        testContext = new TestContext("mapping-multi-field-variation.json");
+    }
+
+    private final String name;
+    private final String query;
+    private final List<Matcher<String>> matchers;
+
+    public QueryTranslatorSpecTests(String name, String query, List<Matcher<String>> matchers) {
```

Comment:
Do we have tests where the same query is ran against more than one matcher?

Reply:
yep sure, please check: https://github.com/elastic/elasticsearch/pull/71829/files#diff-ae8ba179d732e7ad13176e9c0645cc9b5a26b0c9300fe8edbadf22648e334507R138 for example.